lib/validator.js

```diff
@@ -258,6 +257,46 @@ Specberus.prototype.getSotDSection = function () {
     return this.sotdSection;
 };
 
+Specberus.prototype.extractHeaders = function () {
+  var $ = this.$
+  ,   self = this
+  ,   $dl = $("body div.head dl").first()
+  ,   dts = {}
+  ,   EDITORS_DRAFT = /Latest\ editor's\ draft/i;
+  $dl.find("dt").each(function (idx) {
+    var $dt = $(this)
+    ,   txt = self.norm($dt.text())
+                  .replace(":", "")
+                  .toLowerCase()
+                  .replace("published ", "")
+    ,   $dd = $dt.next('dd')
+    ,   key = null
+    ;
+    if (!$dd) return err("no-dd", { dt: txt });
+    if (txt === "this version") key = "This";
+    else if (txt.lastIndexOf("latest version", 0) === 0) key = "Latest";
+    else if (/^previous version(?:s)?$/.test(txt)) key = "Previous";
+    else if (/^rescinds this recommendation?$/.test(txt)) key = "Rescinds";
+    if (EDITORS_DRAFT.test(txt) && $dd.find('a')) key = "Editor";
+    if (key) dts[key] = { pos: idx, el: $dt, dd: $dd };
+  });
+  return dts;
+};
+
+Specberus.prototype.getEditorIDs = function () {
```

Comment:
If the logic is moved in a rule that is used in the `metadata` profile, then [Specberus.validate](https://github.com/w3c/specberus/blob/master/lib/validator.js#L74) relies on a preliminary run of the metadata extraction which is not needed in some situations.
Also, `validator.js` is already getting other data such as the SotD or the document date.